sdk/python/tests/compiler/component_builder_test.py

```diff
@@ -49,6 +52,56 @@ def test_upload_gcs_path(self):
     os.remove(temp_file)
     os.remove(temp_downloaded_file)
 
+class TestDependencyVersion(unittest.TestCase):
+
+  def test_version(self):
+    """ test version overrides min_version and max_version """
+    version = DependencyVersion(version='0.3.0', min_version='0.1.0', max_version='0.4.0')
+    self.assertTrue(version.min_version == '0.3.0')
+    self.assertTrue(version.max_version == '0.3.0')
+    self.assertTrue(version.has_versions())
+    self.assertTrue(version.has_min_version())
+    self.assertTrue(version.has_max_version())
+
+  def test_minmax_version(self):
```

Comment:
Please also check for min only and max only.

Reply:
Good idea. Added.

Reply:
If you implement https://github.com/kubeflow/pipelines/pull/219/commits/000ca3ed9906f1706227462236db5a4ab37eaedc#r233251201 you'll see why I asked for these tests.